libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -325,6 +325,16 @@ void ContractCompiler::appendCalldataUnpacker(TypePointers const& _typeParameter
 {
 	// We do not check the calldata size, everything is zero-padded
 
+	TODO: What about external functions?
+	if (m_context.experimentalFeatureActive(ExperimentalFeature::ABIEncoderV2))
```

Comment:
`ABIDecoderV2`?

Reply:
Perhaps we should have named it differently. Do you want to have a second experimental feature name?

Reply:
It is a different feature, isn't? Or do you think they would mature the same rate? I mean the question is: would you consider to promote one without the other out of experimental status or would prefer to only promote them together.

Reply:
The main feature is being able to call a function on a different contract and pass structs and nested arrays. You always need both the encoder and the decoder.

I also think that the decoder should be easier to write, because you do not have to deal with storage.